File: cpp/1051_HeightChecker.cpp
class Solution {
public:
    int heightChecker(vector<int>& heights) {
        vector<int> unsorted_heights;
        copy(heights.begin(), heights.end(), back_inserter(unsorted_heights));
        
        sort(heights.begin(), heights.end(), [](const int& a, const int& b) {
            return a < b;
        });
        
        int count = 0;
        for (int i = 0; i < heights.size(); ++i) {
            if (heights[i] != unsorted_heights[i])
                ++count;
        }
        return count;
    }
};

File: swift/1051_HeightChecker.swift
class Solution {
    func heightChecker(_ heights: [Int]) -> Int {
        var sortedHeights: [Int] = heights.sorted(by: {
            a, b in
            return a < b
        })
        var count: Int = 0, N: Int = heights.count
        var idx = heights.index(heights.startIndex, offsetBy: 0) // refers to sortedHeights idx as well
        
        for _ in 0..<N {
            if sortedHeights[idx] != heights[idx] { count += 1 }
            sortedHeights.formIndex(after: &idx)
        }
        return count
    }
}
